### PR TITLE
Time regions support in graph panel

### LINKED ISSUE #5930: [Feat Req]: Highlight and filter out weekends or off work hours in graphs
- **I'm submitting a ...**
- [ ] Bug report
- [x] Feature request
- [ ] Question / Support request: **Please do not** open a github issue. [Support Options](http://grafana.org/support/)

I would be nice to have a setting that would enable showing weekends in the graphs with a slightly different background (lighter or darker one).

### PR DESCRIPTION
**Pull request by marefr**
Fixes #5930 

Example dashboard using Grafana Testdata datasource: https://gist.github.com/marefr/2fee183a7b566ab12c6477fccec5a887

image

image

image

image

### CODE REVIEW

**Comment by torkelo** [comment]
I wonder how much conflict this will have with the major refactorings in https://github.com/grafana/grafana/pull/12060/files

**Comment by marefr** [comment]
Good question :)

**Comment by torkelo** [comment]
Maybe thresholds are the wrong name for this? Been thinking about thresholds recently and how we can align the thresholds model  & UI across panels. But this feels a lot different. more like `Colored Time Regions` or Time Markings.

**Comment by marefr** [comment]
Yes agree, but you mean thresholds in general? Colored Time Regions or Time Markings are exactly the same as Colored Value Regions or Value Markings, right?

**Comment by marefr** [comment]
@torkelo did some work on this:
* No longer combined with thresholds and moved to a separate section under Display tab
* The whole concept/feature is now called "time regions"

**Comment by perplexityjeff** [comment]
This is a great feature that we are looking for to be added to Grafana. Is there any update for this request at this moment?

**Comment by marefr** [comment]
@perplexityjeff hopefully it can be included in v5.4.

**Comment by StephenOTT** [comment]
Any plans to add this into other panel types such as HeatMap? thinking prometheus histogram heatmap usage as a example.

**Comment by marefr** [comment]
@StephenOTT not at the moment. That panel is using d3 so it's not that easy to implement just by copy this solution. Please open a separate issue for heatmap time regions.

File: docs/sources/features/panels/graph.md [hunk 1]
```diff
@@ -186,6 +186,14 @@ There is an option under Series overrides to draw lines as dashes. Set Dashes to
 Thresholds allow you to add arbitrary lines or sections to the graph to make it easier to see when
 the graph crosses a particular threshold.
 
+### Time Regions
+
+> Only available in Grafana v5.4 and above.
+
+{{< docs-imagebox img="/img/docs/v54/graph_time_regions.png" max-width= "800px" >}}
```
Comment (torkelo) [inline docs/sources/features/panels/graph.md]: Is this image in the grafana.org repo?

**Review by torkelo — APPROVED** [review]
Looks really great. Sorry for taking for every to review & test this. 

I would remove the background %3 & %6 and rename background %9 with Gray. And maybe add a blue transparent color as well.

File: docs/sources/features/panels/graph.md (hunk 1 above)
Comment (marefr) [reply docs/sources/features/panels/graph.md]: Yes in description

> PR for docs screenshot: grafana/grafana.org#77

**Comment by marefr** [comment]
Replaced background color modes with Gray. Added blue color. Moved custom to bottom.

image

File: docs/sources/features/panels/graph.md (hunk 1 above)
Comment (marefr) [reply docs/sources/features/panels/graph.md]: Will push that image to master directly now.

**Comment by Gauravshah** [comment]
Are there plans to support timezone ? if there are some directions we would love to contribute

**Comment by marefr** [comment]
@Gauravshah what do you mean? You configure times that correlates to UTC. Changing timezone of dashboard to local browser time will adjust the time regions accordingly.

**Comment by Gauravshah** [comment]
@marefr lets say my timezone is PT ( Los Angeles ). I select `Sat` as my time region.  It would show on my graph as Friday night 8 PM to Saturday 8 PM. Then lets say then I pick `Sat 8 AM` to `Sun 8:AM` which will correctly show as Saturday PT. But this would go off when we change DST. We would need to go back and fix all dashboards from 8 AM to 7 AM

**Comment by marefr** [comment]
I would say then use UTC time in dashboard as timezone instead or open a feature request issue.

**Comment by Gauravshah** [comment]
@marefr I think there are valid uses cases around using time zone. Will open a feature request.

**Pull request merged**